include/envoy/grpc/async_client_manager.h

```diff
@@ -42,6 +42,12 @@ class AsyncClientManager {
   virtual AsyncClientFactoryPtr
   factoryForGrpcService(const envoy::config::core::v3::GrpcService& grpc_service,
```

Comment:
If we want to enforce caching this factory method should be better removed from the public interface.

Reply:
My concern is: do we want to enforce caching of async client? I prefer to leave the choice not to use cache.

Reply:
Are there downsides of using the cache?

Reply:
@htuch what's your take? Should we expose an API for instantiating without a cache?

Reply:
We should always cache IMHO, or folks will get unexpected brokeness. There are downside to caching if we don't have eviction, but we should fix that inside the caching implementation, not at individual sites like ext_authz.

Reply:
Thanks, I will make the factory method private.